lib/mongoid/association/depending.rb

```diff
@@ -131,7 +131,9 @@ def _dependent_restrict_with_exception!(association)
 
       def _dependent_restrict_with_error!(association)
         if (relation = send(association.name)) && !relation.blank?
-          errors.add(association.name, RESTRICT_ERROR_MSG)
+          association_name_translated = ::I18n.translate("activerecord.models.#{association.name}", default: association.name.to_s)
+
+          errors.add(:base, :restrict_with_error_dependent_destroy, association: association_name_translated)
```

Comment:
Why is the error added to base instead of the association? As far as I can tell the error is on an object in the association and not on the base model.